### PR TITLE
update VideoSummary, LiveData, llama-on-prem for Llama 3

### PR DESCRIPTION
**Pull request by jeffxtang**
# What does this PR do?

VideoSummary: show how to ask Llama 3 to generate a summary of a long (almost 3 hours) youtube video (Yann LeCun vs Lex Fridman) content with 140+k chars or ~30k words or 40k tokens, exceeding Llama 3's 8k context length limit.
https://github.com/meta-llama/llama-recipes/blob/demos4llama3v3/recipes/use_cases/VideoSummary.ipynb

LiveData: How to ask Llama 3 questions about recent live data via the live search API and LlamaIndex
https://github.com/meta-llama/llama-recipes/blob/demos4llama3v3/recipes/use_cases/LiveData.ipynb

llama-on-prem.md: https://github.com/meta-llama/llama-recipes/blob/demos4llama3v3/recipes/inference/model_servers/llama-on-prem.md

Fixes # (issue)

## Feature/Issue validation/testing

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?  
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

**Comment by albertodepaola** [comment]
Link to the actual recipe, for easier parsing: https://github.com/meta-llama/llama-recipes/blob/demos4llama3v3/recipes/use_cases/VideoSummary.ipynb

**Comment by albertodepaola** [comment]
nit: would be good to add the open in colab style link, from other notebooks as well.

File: recipes/use_cases/LiveData.ipynb [hunk 1]
```diff
@@ -32,37 +19,27 @@
...
    "metadata": {},
    "source": [
-    "Next we set up the Replicate token."
+    "You will be using [Replicate](https://replicate.com/meta/meta-llama-3-8b-instruct) to run the examples here. You will need to first sign in with Replicate with your github account, then create a free API token [here](https://replicate.com/account/api-tokens) that you can use for a while. You can also use other Llama 3 cloud providers such as [Groq](https://console.groq.com/), [Together](https://api.together.xyz/playground/language/meta-llama/Llama-3-8b-hf), or [Anyscale](https://app.endpoints.anyscale.com/playground) - see Section 2 of the Getting to Know Llama [notebook](https://github.com/meta-llama/llama-recipes/blob/main/recipes/quickstart/Getting_to_know_Llama.ipynb) for more information.\n",
+    "\n",
+    "If you'd like to run Llama 3 locally for the benefits of privacy, no cost or no rate limit (some Llama 3 hosting providers set limits for free plan of queries or tokens per second or minute), see [Running Llama Locally](https://github.com/meta-llama/llama-recipes/blob/main/recipes/quickstart/Running_Llama2_Anywhere/Running_Llama_on_Mac_Windows_Linux.ipynb)."
```
Comment (fbnav) [inline recipes/use_cases/LiveData.ipynb:36]: Should we also mention the option of running it using raw weights from our llama3 repo or HF for added benefits of privacy if their setup allows enough VRAM?

**Comment by fbnav** [comment]
Looks great, thanks for putting this together!

File: recipes/use_cases/LiveData.ipynb (hunk 1 above)
Comment (jeffxtang) [reply recipes/use_cases/LiveData.ipynb:36]: I can we can update the main/demo readme by providing a flowchart to help people navigate and choose different notebooks/options for different use cases.

**Review by albertodepaola — CHANGES_REQUESTED** [review]
Adde some inline comments where I could find the relevant line to comment on.

File: recipes/use_cases/VideoSummary.ipynb [hunk 2]
```diff
@@ -20,10 +22,7 @@
...
     "- [langchain](https://python.langchain.com/docs/get_started/introduction) provides necessary RAG tools for this demo\n",
     "- [tiktoken](https://github.com/openai/tiktoken) BytePair Encoding tokenizer\n",
-    "- [pytube](https://pytube.io/en/latest/) Utility for downloading YouTube videos\n",
-    "\n",
-    "**Note** This example uses Replicate to host the Llama model. If you have not set up/or used Replicate before, we suggest you take a look at the [HelloLlamaCloud](HelloLlamaCloud.ipynb) example for information on how to set up Replicate before continuing with this example.\n",
-    "If you do not want to use Replicate, you will need to make some changes to this notebook as you go along."
+    "- [pytube](https://pytube.io/en/latest/) Utility for downloading YouTube videos"
    ]
   },
   {
```
[the remark this answers is not in the document]
Comment (jeffxtang) [reply recipes/use_cases/VideoSummary.ipynb:28]: thanks added

File: recipes/use_cases/VideoSummary.ipynb [hunk 3]
```diff
@@ -168,190 +139,92 @@
...
-      "\u001b[0;31mRuntimeError\u001b[0m: Your input is too long. Max input length is 4096 tokens, but you supplied 5608 tokens."
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
-    "from langchain.chains.summarize import load_summarize_chain\n",
```
[the remark this answers is not in the document]
Comment (jeffxtang) [reply recipes/use_cases/VideoSummary.ipynb:296]: added

File: recipes/use_cases/VideoSummary.ipynb [hunk 4]
```diff
@@ -360,52 +233,52 @@
...
+   "source": [
+    "The `refine` type implements the following steps under the hood:\n",
+    "\n",
+    "1. Call Llama 3 on the first sub-document to generate a concise summary;\n",
+    "2. Loop over each subsequent sub-document, pass the previous summary with the current sub-document to generate a refined new summary;\n",
+    "3. Return the final summary generated on the final sub-document as the final answer - the summary of the whole content.\n",
+    "\n",
+    "An example prompt template for each call in step 2 is:\n",
+    "```\n",
+    "Your job is to produce a final summary.\n",
```
[the remark this answers is not in the document]
Comment (jeffxtang) [reply recipes/use_cases/VideoSummary.ipynb:265]: it's used by LC automatically. showing the example to explain a bit. but good Q - added some clarification.

**Review by albertodepaola — APPROVED** [review]
Looks great, thanks for the quick update!

**Pull request merged**